be/src/vec/core/block.h

```diff
@@ -104,6 +104,16 @@ class Block {
                                      doris::vectorized::IColumn* input_col_ptr,
                                      uint16_t* sel_rowid_idx, uint16_t select_size, int block_cid,
                                      size_t batch_size) {
+        // Only the additional deleted filter condition need to materialize column be at the end of the block
+        // We should not to materialize the column of query engine do not need. So here just return OK.
+        // Eg:
+        //      `delete from table where a = 10;`
+        //      `select b from table;`
+        // a column only effective in segment iterator, the block from query engine only contain the b column.
+        // so the `block_cid >= data.size()` is true
+        if (block_cid >= data.size())
```

Comment:
After `_output_column_by_sel_idx` is executed, `Block::shrink_char_type_column_suffix_zero` may be executed, so it may be necessary to judge, for example

```cpp
void Block::shrink_char_type_column_suffix_zero(std::vector<size_t> char_type_idx) {
    for (auto idx : char_type_idx) {
        if (idx >= data.size()) {
            continue;
        }
        // ...
    }
}
```

Reply:
OK，I will fix it